polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -870,6 +870,26 @@ ValueCategory MLIRScanner::VisitConstructCommon(clang::CXXConstructExpr *Cons,
 
 ValueCategory
 MLIRScanner::VisitArraySubscriptExpr(clang::ArraySubscriptExpr *Expr) {
+  LLVM_DEBUG({
+    llvm::dbgs() << "VisitArraySubscriptExpr: ";
+    Expr->dump();
+    llvm::dbgs() << "\n";
+  });
+
+  assert(!Expr->getBase()->getType()->isVLSTBuiltinType() &&
+         "Not supported yet");
+
+  if (Expr->getBase()->getType()->isVectorType()) {
+    // Handle the vector case.
+    ValueCategory Base{Visit(Expr->getBase()).getValue(Builder), false};
+    auto Idx = Visit(Expr->getIdx());
+
+    mlirclang::warning() << "Not emitting bounds check\n";
```

Comment:
I'm wondering whether we really need this warning. Is the subscript expression is out of bounds the program is incorrect.  I do not think we should generate any code to check that the index expression is within the declared array bounds, with that in mind I think we can remove this kind of warnings.

Reply:
If `clang` is passed `-fsanitize=bounds`, the code generated actually differs, checking that the index is inbounds (for variable indices). That's why I'm adding this kind of warnings. As we might want to add this functionality in the future, I'd say we want to keep this for now. Do you agree?

Reply:
Yes we can keep it for now, in a future PR we should then emit this warning only when `-fsanitize=bounds` is set.